examples/ping/src/icmpv4.adb

```diff
@@ -0,0 +1,271 @@
+      type Octet_Address is array (1 .. 4) of Octet;
+      package Val is new Basalt.Strings_Generic.Value_Option_Modular (Octet);
+      Address : Octet_Address := (others => 0);
+      Oct_First : Natural;
+      Oct_Last  : Natural;
+      Oct_Index : Positive := Address'First;
+      V         : Val.Optional;
+   begin
+      Addr  := 0;
+      Valid := False;
+      if Str'First not in Positive'Range then
+         return;
+      end if;
+      Oct_First := Str'First;
+      Oct_Last  := Str'First - 1;
+      for I in Str'Range loop
+         pragma Loop_Invariant (Oct_First in Str'Range);
+         pragma Loop_Invariant (Oct_Last <= Str'Last);
+         pragma Loop_Invariant (Oct_Index in Address'Range);
+         case Str (I) is
+            when '0' .. '9' =>
+               Oct_Last := I;
+               if I = Str'Last then
+                  V := Val.Value (Str (Oct_First .. Oct_Last));
+                  if not V.Valid or Oct_Index /= Address'Last then
+                     return;
+                  end if;
+                  Address (Oct_Index) := V.Value;
+               end if;
+            when '.' =>
+               V := Val.Value (Str (Oct_First .. Oct_Last));
+               if not V.Valid then
+                  return;
+               end if;
+               Address (Oct_Index) := V.Value;
+               if I = Str'Last then
+                  return;
+               end if;
+               Oct_First := I + 1;
+               Oct_Last  := I;
+               if Oct_Index = Address'Last then
+                  return;
+               end if;
+               Oct_Index := Oct_Index + 1;
+            when others =>
+               return;
+         end case;
+      end loop;
+      for I in Address'Range loop
+         Addr := Addr + RFLX.IPv4.Address (Int.Shift_Left (Int.Unsigned_32 (Address (I)), (I - 1) * 8));
+      end loop;
+      Valid := True;
+   end Get_Address;
+
+   --------------
+   -- Generate --
+   --------------
+
+   procedure Generate (Buf  : in out RFLX.RFLX_Builtin_Types.Bytes_Ptr;
+                       Addr :        RFLX.IPv4.Address) is
+      use type RFLX.ICMP.Sequence_Number;
+      use type RFLX.RFLX_Builtin_Types.Bit_Length;
+      IP_Context : RFLX.IPv4.Packet.Context;
+      ICMP_Context : RFLX.ICMP.Message.Context;
+      Data : constant RFLX.RFLX_Builtin_Types.Bytes (1 .. 8) := (others => 65);
+      function Valid_Length (L : RFLX.RFLX_Builtin_Types.Length) return Boolean is
+         (L = Data'Length);
+      procedure Process_Data (Buffer : out RFLX.RFLX_Builtin_Types.Bytes) with
+         Pre => Valid_Length (Buffer'Length);
+      procedure Process_Data (Buffer : out RFLX.RFLX_Builtin_Types.Bytes)
+      is
+      begin
+         Buffer := Data;
+      end Process_Data;
+      procedure Set_Data is new RFLX.ICMP.Message.Set_Bounded_Data (Process_Data, Valid_Length);
+   begin
+      pragma Warnings (Off, "unused assignment to ""*_Context""");
```

Comment:
I suppose this is caused by `Take_Buffer`. Is there a way to remove this warning without suppressing it? For a similar issue we were able to prevent this warning by adding a flow dependency: c86144f5ac11671b931124a0d9c4a00bfcf4d075.

Reply:
I can't tell how I would add a dependency that fixed this in my case. `Buf` does not depend on any other visible parameter and nothing else depends on `Buf`. It is only passed as a working space do avoid reallocations.

Reply:
That warning is about `*_Context`. I suppose it is caused because the context is not used after `Take_Buffer`. My question aimed at a general solution. Maybe we should create an issue for that.

Reply:
I see. We have this issue in Gneiss, too and I'm not aware of a solution.

Reply:
Cf. #368.